### PR TITLE
Tribal totem

### PR DESCRIPTION
**Pull request by twinkle-is-dum**
I should probably rename CombinationLockStep to PuzzleStep but the rest should be good. I also need to see if checkstyle is good. Just opening the PR for now (:

### CODE REVIEW

File: src/main/java/com/questhelper/quests/tribaltotem/CombinationLockStep.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+            highlightButtons.put(5, 0);
+        }
+    }
+
+    private int matchStateToSolution(int slot, String target, int arrowRightId, int arrowLeftId)
+    {
+        Widget widget = client.getWidget(369, slot);
+        if (widget == null) return 0;
+        String current = widget.getText();
+        int Id = current.charAt(0) < target.charAt(0) ? arrowRightId : arrowLeftId;
```
Comment (Zoinkwiz) [inline src/main/java/com/questhelper/quests/tribaltotem/CombinationLockStep.java:97]: You could make this choose to go backwards/forwards considering proximity in both directions by doing something like the following:

```
char current = widget.getText().charAt(0);
int currentPos = (int)current - (int)'A';
int id = Math.floorMod(currentPos - target, 26) < Math.floorMod(target - currentPos, 26) ? arrowRightId : arrowLeftId;
```

File: src/main/java/com/questhelper/quests/tribaltotem/TribalTotem.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+        reqs.add(ardougneTeleports);
+        return reqs;
+    }
+
+    public void loadZones()
+    {
+        houseGroundFloorEntrance = new Zone(new WorldPoint(2637, 3320, 0), new WorldPoint(2639, 3325, 0));
+        houseGroundFloorMiddleRoom = new Zone(new WorldPoint(2634, 3322, 0), new WorldPoint(2636, 3324, 0));
+        houseGroundFloor = new Zone(new WorldPoint(2629, 3321, 0), new WorldPoint(2633, 3325, 0));
+        houseFirstFloor = new Zone(new WorldPoint(2630, 3319, 1), new WorldPoint(2638, 3323, 1));
```
Comment (Zoinkwiz) [inline src/main/java/com/questhelper/quests/tribaltotem/TribalTotem.java:115]: The first floor needs to extend a tile further south and east to capture the entire first floor.

**Review by Zoinkwiz — CHANGES_REQUESTED** [review]
Couple comments otherwise looks good!

File: src/main/java/com/questhelper/quests/tribaltotem/TribalTotem.java [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+
+    public void setupSteps()
+    {
+        talkToKangaiMau = new NpcStep(this, NpcID.KANGAI_MAU, new WorldPoint(2794, 3182, 0), "Talk to Kangai Mau in the Brimhaven food store.");
+        talkToKangaiMau.addDialogSteps("I'm in search of adventure!", "Ok, I will get it back.");
+
+        investigateCrate = new ObjectStep(this, ObjectID.CRATE_2707, new WorldPoint(2650, 3273, 0), "Travel to the GPDT depot in Ardougne and investigate the most northeastern crate for a label.");
+        useLabel = new ObjectStep(this, ObjectID.CRATE_2708, new WorldPoint(2650, 3271, 0), "Use the label on the highlighted crate.", addressLabel);
+        useLabel.addIcon(ItemID.ADDRESS_LABEL);
+        talkToEmployee = new NpcStep(this, NpcID.GPDT_EMPLOYEE, new WorldPoint(2647, 3272, 0), "Talk to a nearby GPDT employee.");
```
[the remark this answers is not in the document]
Comment (twinkle-is-dum) [reply src/main/java/com/questhelper/quests/tribaltotem/TribalTotem.java:139]: Did not know this was possible, added

**Review by Zoinkwiz — APPROVED** [review]
LGTM, thanks!

**Pull request merged**